### ISSUE TITLE
Split DNS support

### ISSUE THREAD

**Issue by davidnortonjr** [tailscale/terraform-provider-tailscale#149]
**Is your feature request related to a problem? Please describe.**

I need to be able to set up a **Restricted Nameserver** or Split DNS for a particular domain

**Describe the solution you'd like**

Something like the following

```
resource "tailscale_dns_restricted_nameserver" "example_com" {
  search_domain = "example.com"
  nameserver = ["1.2.3.4"]
}
```

**Context**

[DNS in Tailscale docs](https://tailscale.com/kb/1054/dns/#using-dns-settings-in-the-admin-console)

**Comment by DentonGentry** [comment]
There is not yet a public API to set Split DNS nameservers: https://github.com/tailscale/tailscale/blob/main/api.md#tailnet-dns-nameservers-post

The public API was blocked by the need to remove the use of fallback resolvers: https://github.com/tailscale/tailscale/issues/1743
A control server update to remove fallback resolvers deployed one week ago.

After the API is updated, the Terraform provider can be updated to add Split DNS support.

**Comment by DentonGentry** [comment]
There is not yet a public API to set Split DNS nameservers. A feature request to do so is tracked in https://github.com/tailscale/tailscale/issues/5932

**Comment by gberenice** [comment]
Hey! 
Would love to see this feature implemented! I see that the issue https://github.com/tailscale/tailscale/issues/5932 is closed. Are there any estimations about tf provider update?

**Comment by DentonGentry** [comment]
https://github.com/tailscale/tailscale/issues/5932 was closed as a duplicate of https://github.com/tailscale/tailscale/issues/2661, which is still open. There is not currently a public API to set Split DNS nameservers.

**Comment by gberenice** [comment]
@DentonGentry ah, you're right. Thanks, looking forward to further updates.

**Comment by mrgasparov** [comment]
+1

**Comment by mrgasparov** [comment]
Any updates? Would be awesome to not have to set these by hand 😁 Thanks

**Comment by aiell0** [comment]
Would love a solution here if anyone has one. It's frustrating to have this be the only manual step in my Terraform at this point.

**Comment by mpminardi** [comment]
This is resolved as of #359 . A new [tailscale_dns_split_nameservers](https://registry.terraform.io/providers/tailscale/tailscale/latest/docs/resources/dns_split_nameservers) resource has been added as of the [v0.16.0 ](https://github.com/tailscale/terraform-provider-tailscale/releases/tag/v0.16.0)release.
